packages/connection-model/package.json

```diff
@@ -15,7 +15,8 @@
   ],
   "scripts": {
     "test-check-ci": "npm run check && npm test",
-    "pretest": "mongodb-runner install",
+    "prepretest": "mongodb-runner install",
+    "pretest": "node ../../scripts/rebuild.js keytar",
```

Comment:
FWIW I was adding rebuild scripts only in post step for test suites that are running tests with electron runtime. Doesn't hurt to have it here, but might be a bit redundant

Reply:
We really need to do the dynamic package that supports both electron and node when we have some time. The problem is that when you "develop" you'd end up mixing `npm run start` with CMD+R reloads (that does not rebuild) with `npm run test`, and you'd get basically skewed rebuilds that breaks the execution all the times. Since NAPI won't help till 2024 i guess the only option left is the double build trick that gets selected dinamically.

Reply:
I'd say it really depends on ones workflow then, I don't often jump between running tests and running an app so haven't noticed this an an issue. But to make it clear I am not dismissing this as an issue, let's open a ticket? We can also circumvent this in other ways, it's just if we will start adding `rebuild` before `npm run test` in more packages, it might slow down the tests (and CI flows) quite a bit

Reply:
Oh, also will add this to the docs

Reply:
yeah i can create a ticket, the rebuild before tests should be a good idea, i bet there is a way to "detect" if a rebuild is required. 

Another thing we could do is "flip" the logic. Right now we build for node and we keep it built for electron only while electron is running. We could do the other way around (build for electron by default) and only rebuild for node while running the tests.

After a while you get used to it but that issue is a bit of a deep thing to circumvent if you get it and don't have enough experience, i'm thinking about external or casual contributors / newbies too, and in general a random source of time wasted. 

That's why I feel quite good about that "dynamic" require trick if we could timebox it to 1-2 days and would solve the issue forever. If it takes more or is problematic otherwise we could give up and fallback to rebuilds.

Reply:
> Another thing we could do is "flip" the logic. Right now we build for node and we keep it built for electron only while electron is running. We could do the other way around (build for electron by default) and only rebuild for node while running the tests.

We can switch all tests to use `electron-mocha` as a test runner (a lot of packages are using it already) and then you don't need to rebuild anything back to node env at all I guess

Reply:
> We can switch all tests to use electron-mocha as a test runner (a lot of packages are using it already) and then you don't need to rebuild anything back to node env at all I guess

Ohh that i would like!! :)